packages/expect/src/__tests__/spyMatchers.test.ts

```diff
@@ -841,72 +992,108 @@ describe.each([
   'toReturnWith',
   'toHaveReturnedWith',
 ] as const)('%s', returnedWith => {
-  const caller = function (
-    callee: (...a: Array<unknown>) => void,
-    ...args: Array<unknown>
-  ) {
-    if (
+  function isNth(
+    returnedWith: string,
+  ): returnedWith is 'nthReturnedWith' | 'toHaveNthReturnedWith' {
```

Comment:
I was trying to invent less verbose abstraction. Was not happy about it ;D The test is huge and somehow readability won in the end.